src/main/java/fi/aalto/cs/apluscourses/model/SubmissionStatusUpdater.java

```diff
@@ -45,14 +49,16 @@
   /**
    * Construct a submission status updater with the given parameters.
    */
-  public SubmissionStatusUpdater(@NotNull ExerciseDataSource dataSource,
+  public SubmissionStatusUpdater(@NotNull Project project,
```

Comment:
I would perhaps make this `@Nullable` instead, since `ActionEvent#getProject` is also `@Nullable`. Currently of course `SubmitExerciseAction` only works if the project isn't null, but since `SubmissionStatusUpdater` only uses the project for the notification, I don't think it should require it to be non-null.

Reply:
Good point. Should I leave SubmissionStatusUpdaterTest as is?

Reply:
Yep the mock project is fine :+1: